WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -963,6 +965,18 @@ public void onAccountChanged(OnAccountChanged event) {
         }
     }
 
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPlansFetched(OnPlansFetched event) {
+        if (event.isError()) {
+            AppLog.e(T.DOMAIN_REGISTRATION, "An error occurred while fetching plans : " + event.error.message);
+        } else {
+            MySiteFragment fragment = getMySiteFragment();
+            if (fragment != null) {
+                fragment.updateDomainRegistrationCtaIfDomainCreditAvailable(event.plans);
```

Comment:
`updateDomainRegistrationCtaIfDomainCreditAvailable` changes the visibility of `MySiteFragment` views, but fragment being not null, does not guarantee that its View is created and can be manipulated. In this particular case, `fragment` will never be null (probably)  - if you go deeper into `getMySiteFragment()` you will see that it's lazy-loading `MySiteFragment`. This will cause an unrecoverable crash (you will have to reinstall the app to fix it) if you switch to another tab (from the bottom nav bar) and restart the app.

Reply:
Great catch. I have addressed this issue in the latest round of changes. I only call `fetchPlansIfNecessary` from `onResume`. It _seems_ to catch all cases where the current site has been changed. I was originally trying to call it in `onSiteChanged` as well, but it creates the same issue since `onSiteChanged` is called using `getMySiteFragment`.